### PR TITLE
Adding basic support for HorizontalPodAutoscaler resources from autoscalingv2Beta1

### LINKED ISSUE #243: [FEATURE_REQUEST] add support to check horizontal pod autoscaler
you could add a check or template for validate horizontal pod autoscaler files.

### PR DESCRIPTION
**Pull request by trumant**
Closes #243

Curious to hear thoughts on whether support should also be added for autoscalingv1  and/or autoscalingv2Beta2

Also, as this is my first PR to the project, I'm sure I probably missed something obvious. Hoping to learn from your review comments.

### CODE REVIEW

**Comment by janisz** [comment]
@trumant Thanks for PR. I think you need to run `make generated-srcs`

File: e2etests/bats-tests.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 load bats-support-clone
 load test_helper/bats-support/load
-load test_helper/redhatcop-bats-library/src/error-handling.bash
+load test_helper/redhatcop-bats-library/src/error-handling
```
[the remark this answers is not in the document]
Comment (trumant) [reply e2etests/bats-tests.sh:5]: See commit message on https://github.com/stackrox/kube-linter/pull/271/commits/a75c44ae2218b82737358472f1138b328d8f2180

**Comment by trumant** [comment]
> @trumant Thanks for PR. I think you need to run `make generated-srcs`

Done!

**Review by viswajithiii — APPROVED** [review]
This is great, thanks @trumant

**Comment by viswajithiii** [comment]
Hey @trumant, it looks like the test you wrote is failing in CI... Can you take a look?

**Comment by trumant** [comment]
@viswajithiii Tests should now be passing, please approve the CI workflows

**Review by janisz — APPROVED** [review]

**Comment by janisz** [comment]
@trumant Merged 🎉 Thanks!

**Pull request merged**